src/lib/state-operators/removal.ts

```diff
@@ -0,0 +1,91 @@
+import { StateOperator } from '@ngxs/store';
+import { compose, patch } from '@ngxs/store/operators';
+import { Predicate } from '@ngxs/store/operators/internals';
+import { Dictionary } from '../internal';
+import { EntityStateModel } from '../models';
+import { updateTimestamp } from './timestamp';
+
+/**
+ * Removes all entities, clears the active entity and updates the `lastUpdated` timestamp.
+ */
+export function removeAllEntities<T>(): StateOperator<EntityStateModel<T>> {
+  return (state: EntityStateModel<T>) => {
+    return {
+      ...state,
+      entities: {},
+      ids: [],
+      active: undefined,
+      lastUpdated: Date.now()
+    };
+  };
+}
+
+/**
+ * Removes the entities specified by the given IDs.
+ * The active entity will be cleared if included in the given IDs.
+ * Updates the `lastUpdated` timestamp.
+ * @param ids IDs to remove
+ */
+export function removeEntities<T>(ids: string[]): StateOperator<EntityStateModel<T>> {
+  const entityRemoval = patch<EntityStateModel<any>>({
+    entities: removeEntitiesFromDictionary(ids),
+    ids: removeEntitiesFromArray(ids)
+  });
+  return compose(
+    entityRemoval,
+    clearActiveOnRemoval(ids),
+    updateTimestamp()
+  );
+}
+
+/**
+ * Only clears the `active` entity, if it's included in the given array.
+ * All other fields will remain untouched in any case.
+ * @param idsForRemoval the IDs to be removed
+ */
+export function clearActiveOnRemoval<T>(
```

Comment:
Potentially a better name for this (Up to you 😉):
```suggestion
export function clearActiveIfRemoved<T>(
```